packages/fury-adapter-oas3-parser/lib/parser/oas/parseOauthFlowObject.js

```diff
@@ -0,0 +1,67 @@
+const R = require('ramda');
+const {
+  isExtension, hasKey, getValue,
+} = require('../../predicates');
+const {
+  createInvalidMemberWarning,
+} = require('../annotations');
+const pipeParseResult = require('../../pipeParseResult');
+const parseObject = require('../parseObject');
+const parseString = require('../parseString');
+
+const name = 'Oauth Flow Object';
+const requiredKeys = ['scopes'];
+
+/**
+ * Parse Oauth Flow Object
+ *
+ * @param namespace {Namespace}
+ * @param element {Element}
+ * @returns ParseResult
+ *
+ * @see https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#oauthFlowObject
+ * @private
+ */
+function parseOauthFlowObject(context, object) {
+  const { namespace } = context;
+  const scopesName = `${name}' 'scopes`;
+
+  const parseScopeMember = R.cond([
+    [R.T, parseString(context, scopesName, false)],
+  ]);
+
+  const parseMember = R.cond([
+    [hasKey('scopes'), R.compose(parseObject(context, scopesName, parseScopeMember, [], true), getValue)],
+    [hasKey('refreshUrl'), parseString(context, name, false)],
+    [hasKey('authorizationUrl'), parseString(context, name, false)],
+    [hasKey('tokenUrl'), parseString(context, name, false)],
+
+    // FIXME Support exposing extensions into parse result
+    [isExtension, () => new namespace.elements.ParseResult()],
+
+    // Return a warning for additional properties
+    [R.T, createInvalidMemberWarning(namespace, name)],
+  ]);
+
+  const parseOauthFlow = pipeParseResult(namespace,
+    parseObject(context, name, parseMember, requiredKeys, true),
+    (oauthFlow) => {
+      const arr = new namespace.elements.Array([]);
+      const scopes = oauthFlow.get('scopes');
+
+      scopes.forEach((value, key) => {
```

Comment:
This could become a "map" transforming the values instead of accesing and mutating "arr".

```js
oauthFlow.set('scopes', scopes.map((value, key) => {
  const scope = key.clone();
  scope.description = value;
  return scope;
});
```

Or better, "parseScopes" should return an array of the parsed scopes. That way all the logic for parsing the scopes is handled inside the scope parsing and it simplifies the logic here (pseudocode):

```js
const parseScope = pipeParseResult(namespace,
  parseObject(context, scopesName, parseScopeMember, [], true),
  scopes => new namespace.elements.Array(scopes.content),
  R.map(member => {
    const scope = member.key.clone();
    scope.description = member.value;
    return scope;
  }));
```

Then the member parser can be:

```js
[hasKey('scopes'), R.compose(parseScopes, getValue)],
```

That allows the block here to already have the parsed scopes as expected.